client/src/pages/Dashboard.js

```diff
@@ -16,27 +17,60 @@ const useStyles = makeStyles(() => ({
 
 const Dashboard = () => {
   const classes = useStyles();
+  const history = useHistory();
+
+  const [user, setUser] = useState({});
+  const [signedIn, setSignedIn] = useState(false);
+
+  useEffect(async () => {
+    checkCurrentUser();
+  }, []);
+
+  const checkCurrentUser = async () => {
+    //redirect to landing page if no token
+    let results = await fetch("/user/currentUser");
+
+    results = await results.json();
+
+    if (results.user === undefined) {
+      history.push("/");
+    } else {
+      //get user by id
+      setSignedIn(true)
+      let data = await fetch(`/user/${results.user.id}`);
+
+      data = await data.json()
+
+      console.log(data)
+
+      setUser(data.user);
```

Comment:
It's usually not a good idea to overwrite data to parse it. It's better to declare two variables. It makes the code easier to read/understand.
So instead of doing:
```javascript
let data = await fetch(...);
data = await data.json();
```

It's better to do:

```javascript
const data = await fetch(...);
const parsedData = await data.json();
```